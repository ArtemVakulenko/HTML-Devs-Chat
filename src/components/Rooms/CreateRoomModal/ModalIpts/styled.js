import styled from 'styled-components';

export const StIpts = styled.div`
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
  gap: 10px;
`;

export const StFileIpt = styled.label`
  display: inline-block;
  background-color: ${({ colors, theme }) => colors[theme].button};
  color:  ${({ colors, theme }) => colors[theme].buttonText};
  padding: 0.25rem 1rem;
  font-family: sans-serif;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
`;
