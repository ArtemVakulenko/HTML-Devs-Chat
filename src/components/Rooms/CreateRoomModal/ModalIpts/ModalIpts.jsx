import React from 'react';
import PropTypes from 'prop-types';
import Input from 'src/components/_common/Input';
import FileIpt from 'src/components/_common/FileIpt';
import { StIpts } from './styled';

const ModalIpts = ({
  setImg,
  imgName,
  roomValue,
  searchValue,
  setRoomName,
  setSearchUser,
}) => (
  <StIpts>
    <Input
      value={roomValue}
      onChange={setRoomName}
      placeholder='enterChatName'
    />
    <Input
      value={searchValue}
      onChange={setSearchUser}
      placeholder='findUser'
    />
    <FileIpt onChange={setImg} value={imgName} />
  </StIpts>
);

export default ModalIpts;

ModalIpts.propTypes = {
  setImg: PropTypes.func.isRequired,
  imgName: PropTypes.string.isRequired,
  roomValue: PropTypes.string.isRequired,
  setRoomName: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired,
  setSearchUser: PropTypes.func.isRequired,
};
