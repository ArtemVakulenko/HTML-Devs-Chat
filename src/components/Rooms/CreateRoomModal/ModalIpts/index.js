import { connect } from 'react-redux';
import { roomNameValue, searchValue, imgValue } from 'store/modals/selectors';
import { setRoomName, setSearchUser, setImg } from 'store/modals/actions';
import ModalIpts from './ModalIpts';

const mapStateToProps = (state) => ({
  imgName: imgValue(state, 'name'),
  roomValue: roomNameValue(state),
  searchValue: searchValue(state),
});

const mapDispatchToProps = {
  setImg,
  setRoomName,
  setSearchUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(ModalIpts);
