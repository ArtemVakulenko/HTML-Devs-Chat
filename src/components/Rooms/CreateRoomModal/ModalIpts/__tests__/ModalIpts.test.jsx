import { mount, shallow } from 'enzyme';
import React from 'react';
import ModalIpts from '../ModalIpts';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
describe('ModalIpts', () => {
  const props = {
    setImg: jest.fn(),
    imgName: 'sad',
    roomValue: 'sad',
    searchValue: 'sad',
    setRoomName: jest.fn(),
    setSearchUser: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = shallow(<ModalIpts {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StIpts', () => {
    const component = mount(<ModalIpts {...props} />);
    expect(component.find('styled__StIpts')).toHaveLength(1);
  });
  it('Should have Input', () => {
    const component = mount(<ModalIpts {...props} />);
    expect(component.find('Input')).toHaveLength(2);
  });
  it('Should have FileIpt', () => {
    const component = mount(<ModalIpts {...props} />);
    expect(component.find('FileIpt')).toHaveLength(1);
  });
});
