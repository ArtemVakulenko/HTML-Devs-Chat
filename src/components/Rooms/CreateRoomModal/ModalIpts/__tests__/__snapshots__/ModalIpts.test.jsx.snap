// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalIpts Should match snapshot 1`] = `"<div class=\\"styled__StIpts-cEsiqO cVGEcg\\"><input type=\\"text\\" value=\\"sad\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><input type=\\"text\\" value=\\"sad\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><label for=\\"upload\\" class=\\"styled__StFileIpt-fLkDRX iPkIEE\\">sad</label><input type=\\"file\\" id=\\"upload\\" hidden=\\"\\"/></div>"`;
